src/quote/Quote.ts

```diff
@@ -108,3 +149,156 @@ export function createAgreedQuote(
+
+export function decompressCost(cost: CompressedCostBreakdown): ICostBreakdown {
+  return { gross: cost[0], net: cost[1], tax: cost[2] }
+}
+
+/**
+ *  Compresses a [[Quote]] for storage and/or messaging.
+ *
+ * @param quote An [[Quote]] object that will be sorted and stripped for messaging or storage.
+ *
+ * @returns An ordered array of an [[Quote]].
+ */
+
+export function compressQuote(quote: IQuote): CompressedQuote {
+  return [
+    quote.attesterAddress,
+    quote.cTypeHash,
+    compressCost(quote.cost),
+    quote.currency,
+    quote.termsAndConditions,
+    quote.timeframe,
+  ]
+}
+
+/**
+ *  Decompresses an [[Quote]] from storage and/or message.
+ *
+ * @param quote A compressesd [[Quote]] array that is reverted back into an object.
+ *
+ * @returns An object that has the same properties as an [[Quote]].
+ */
+
+export function decompressQuote(quote: CompressedQuote): IQuote {
+  return {
+    attesterAddress: quote[0],
+    cTypeHash: quote[1],
+    cost: decompressCost(quote[2]),
+    currency: quote[3],
+    termsAndConditions: quote[4],
+    timeframe: quote[5],
+  }
+}
+
+/**
+ *  Compresses an attester signed [[Quote]] for storage and/or messaging.
+ *
+ * @param attesterSignedQuote An attester signed [[Quote]] object that will be sorted and stripped for messaging or storage.
+ *
+ * @returns An ordered array of an attester signed [[Quote]].
+ */
+
+export function compressAttesterSignedQuote(
+  attesterSignedQuote: IQuoteAttesterSigned
+): CompressedQuoteAttesterSigned {
+  return [
+    attesterSignedQuote.attesterAddress,
+    attesterSignedQuote.cTypeHash,
+    compressCost(attesterSignedQuote.cost),
+    attesterSignedQuote.currency,
+    attesterSignedQuote.termsAndConditions,
+    attesterSignedQuote.timeframe,
+    attesterSignedQuote.attesterSignature,
+  ]
+}
+
+/**
+ *  Decompresses an attester signed [[Quote]] from storage and/or message.
+ *
+ * @param attesterSignedQuote A compressesd attester signed [[Quote]] array that is reverted back into an object.
+ *
+ * @returns An object that has the same properties as an attester signed [[Quote]].
+ */
+
+export function decompressAttesterSignedQuote(
+  attesterSignedQuote: CompressedQuoteAttesterSigned
+): IQuoteAttesterSigned {
+  return {
+    attesterAddress: attesterSignedQuote[0],
+    cTypeHash: attesterSignedQuote[1],
+    cost: decompressCost(attesterSignedQuote[2]),
+    currency: attesterSignedQuote[3],
+    termsAndConditions: attesterSignedQuote[4],
+    timeframe: attesterSignedQuote[5],
+    attesterSignature: attesterSignedQuote[6],
+  }
+}
+
+/**
+ *  Compresses a [[Quote]] Agreement for storage and/or messaging.
+ *
+ * @param quoteAgreement A [[Quote]] Agreement object that will be sorted and stripped for messaging or storage.
+ *
+ * @returns An ordered array of a [[Quote]] Agreement.
+ */
+
+export function compressQuoteAgreement(
+  quoteAgreement: IQuoteAgreement
+): CompressedQuoteAgreed {
+  return [
+    quoteAgreement.attesterAddress,
```

Comment:
You can reuse `compressAttesterSignedQuote` here.

Also look at the other functions, if you can reuse more functions.